lib/ansible/utils/encrypt.py

```diff
@@ -64,10 +65,14 @@ class BaseHash(object):
     algo = namedtuple('algo', ['crypt_id', 'salt_size', 'implicit_rounds'])
     algorithms = {
         'md5_crypt': algo(crypt_id='1', salt_size=8, implicit_rounds=None),
-        'bcrypt': algo(crypt_id='2a', salt_size=22, implicit_rounds=None),
         'sha256_crypt': algo(crypt_id='5', salt_size=16, implicit_rounds=5000),
         'sha512_crypt': algo(crypt_id='6', salt_size=16, implicit_rounds=5000),
     }
+    system = to_text(platform.system(), errors='surrogate_or_strict')
+    if system == u"OpenBSD":
```

Comment:
I'd probably prefer a single assignment with a conditional only for the argument. It would probably look cleaner this way:
```suggestion
    is_openbsd = u"OpenBSD" == to_text(platform.system(), errors='surrogate_or_strict')
    bcrypt_crupt_id = '2b' if system == u"OpenBSD" else '2a'
    algorithms['bcrypt'] = algo(crypt_id=bcrypt_crupt_id, salt_size=22, implicit_rounds=None)
```